src/cryptsetup/cryptsetup.c

```diff
@@ -1322,8 +1322,11 @@ static int attach_luks_or_plain_or_bitlk_by_tpm2(
                                 return log_error_errno(SYNTHETIC_ERRNO(EAGAIN), "TPM2 PIN unlock failed, falling back to traditional unlocking.");
                         if (ERRNO_IS_NOT_SUPPORTED(r)) /* TPM2 support not compiled in? */
                                 return log_debug_errno(SYNTHETIC_ERRNO(EAGAIN), "TPM2 support not available, falling back to traditional unlocking.");
-                        if (r != -EAGAIN) /* EAGAIN means: no tpm2 chip found */
-                                return r;
+                        /* EAGAIN means: no tpm2 chip found */
+                        if (r != -EAGAIN) {
+                                return log_debug_errno(SYNTHETIC_ERRNO(EAGAIN),
+                                                       "TPM2 operation failed, falling back to traditional unlocking.");
```

Comment:
so this should definitely log the original error code here, and there should be an explicit comment that we mangle error codes here. i.e. something like this:

```c
log_debug_errno(r, TPM2 operation failed, falling back to traditional unlocking: %m");
return -EAGAIN; /* Mangle error code: lLet's make any form of TPM2 failure non-fatal. */
```

Reply:
ok